src/python/pants/backend/python/util_rules/faas.py

```diff
@@ -434,8 +439,32 @@ async def infer_runtime_platforms(request: RuntimePlatformsRequest) -> RuntimePl
         )
     except StopIteration:
         # Not a known runtime, so fallback to just passing a platform
-        # TODO: maybe this should be an error, and we require users to specify
-        # complete_platforms themselves?
+        version_modifier = "[inferred from interpreter constraints]" if inferred_from_ics else ""
+        version_adjective = "inferred" if inferred_from_ics else "specified"
+        known_runtimes_str = ", ".join(
+            FrozenOrderedSet(r.name for r in request.runtime.known_runtimes)
+        )
+        warn_or_error(
+            "2.26.0.dev0",
+            "implicitly resolving platforms for unknown FaaS runtimes",
+            softwrap(
+                f"""
+                Could not find a known runtime for the {version_adjective} Python version and machine architecture!
+
+                Python version: {version} {version_modifier}
+                Machine architecture: {request.architecture.value}
+
+                Known runtime values: {known_runtimes_str}
+
+                To fix, please generate a `complete_platforms` file for the given Python version and
+                machine architecture, or specify a runtime that is known to Pants.
+
+                You can follow the instructions at https://github.com/pantsbuild/pants/discussions/18756
```

Comment:
Let's update this link to the new docs.

Reply:
I meant to do this before but somehow lost it 🤦 

Thanks for the reminder